test/test_cuda.py

```diff
@@ -1051,6 +1052,12 @@ def run(dev: torch.device) -> int:
             abs(run(torch.device("cuda")) - run(torch.device("cpu"))) < 10_000
         )
 
+    @largeTensorTest("20GB", "cuda")
+    def test_randint_generation_for_large_numel(self) -> None:
+        numel = 2**31 + 1
+        s = torch.randint(2, (numel,), device="cuda", dtype=torch.int8).sum()
```

Comment:
I think if you don't do `sum` here you'd be able to get away with less memory, `sum` upcasts to int64. If we are just doing a smoke test, checking `sum` on the first 1K elements should be enough, the probability that it's 0 is 2^(-1000) = 10^(-300), no one should be that unlucky.